luarules/gadgets/unit_shield_behaviour.lua

```diff
@@ -304,8 +302,8 @@ function gadget:ShieldPreDamaged(proID, proOwnerID, shieldWeaponNum, shieldUnitI
 	-- proID isn't nil if hitscan weapons are used, it's actually -1.
 	if proID > -1 then
 		weaponDefID = projectileDefIDCache[proID] or spGetProjectileDefID(proID)
-
-		shieldData.shieldDamage = shieldData.shieldDamage + originalShieldDamages[weaponDefID]
+		local newShieldDamage = originalShieldDamages[weaponDefID] or 0 -- 0 to prevent rare bug where originalShieldDamages[weaponDefID] returns nil
```

Comment:
why not just fill the missing part of the table?

Reply:
Wait a minute, is this from weapons whose damages table contains neither a `shields` entry nor a `default` entry?
https://github.com/beyond-all-reason/Beyond-All-Reason/blob/1284e003b762f6fefa339dff4eb4bb85597018fd/gamedata/alldefs_post.lua#L1396
e.g. AA weapons doing no damage to shields

Reply:
the damage table is supposed to contain entries for all armor classes, even if left unspecified in the def file. the `or default` part should never even be reached

Reply:
alldefs happens before those tables are populated, so it was coming up as nil for vtol damage types that don't have a default defined
I fixed this error.